File: Algorithm-CPP/37_LeastRecentlyUsed.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int s, n;
    cin >> s >> n;

    int arr[s];
    for(int i = 0; i < s; i++) arr[i] = 0;

    for(int i = 0; i < n; i++) {
        int input;
        cin >> input;

        if(find(arr, arr+s, input) != arr+s) {
            int index = 0;
            for(int j = 0; j < s; j++) {
                if(arr[j] == input) {
                    index = j;
                    break;
                }
            }

            for(int j = index - 1; j >= 0; j--) {
                arr[j + 1] = arr[j];
            }
        } else {
            for(int j = s - 2; j >= 0; j--) {
                arr[j + 1] = arr[j];
            }
        }

        arr[0] = input;
    }

    for(int j = 0; j < s; j++) cout << arr[j] << " ";

    return 0;
}

File: Algorithm-Swift/Algorithm-Swift/Inflearn/37_LeastRecentlyUsed.swift
//
//  37_LeastRecentlyUsed.swift
//  Algorithm-Swift
//
//  Created by SHIN YOON AH on 2023/08/16.
//

import Foundation

func solution37() {
    let sizes = readLine()!.split(separator: " ").map { Int($0)! }
    let s = sizes.first!
    let nums = readLine()!.split(separator: " ").map { Int($0)! }
    var arr = Array(repeating: 0, count: s)

    nums.forEach { n in
        if arr.contains(n) {
            let index = arr.firstIndex(of: n)!
            for i in stride(from: index - 1, through: 0, by: -1) {
                arr[i + 1] = arr[i]
            }
        } else {
            for i in stride(from: s - 2, through: 0, by: -1) {
                arr[i + 1] = arr[i]
            }
        }

        arr[0] = n
    }

    arr.forEach { print($0, terminator: " ") }
}
